UnitTestProject1/UnitTestProject1/Task_20.cs

```diff
@@ -0,0 +1,173 @@
+        private static string KeyCode = Convert.ToString('\u0002');
+        private static string AlertMSG = "I am a JS Alert";
+        private static IWebDriver driver;
+        private string ExpectedURL = "https://192.168.100.26/Home/Index";
+        private string OfficeTabURL = "https://192.168.100.26/Office/Index";
+        
+        public void Login(string username, string password)
+        {
+           driver.Navigate().GoToUrl(URL);
+           driver.FindElement(By.CssSelector("input[id=Username]")).SendKeys(username);
+           driver.FindElement(By.Id("Password")).SendKeys(password);
+           driver.FindElement(By.TagName("button")).Click();
+        }
+
+        [TestInitialize]
+        public void initialize()
+        {
+           driver = new ChromeDriver(@"C:\Users\andreygladikov\Source\Repos\NewRepo\UnitTestProject1\UnitTestProject1\UnitTestProject1\SolutionItems");
+           driver.Manage().Timeouts().ImplicitWait=TimeSpan.FromSeconds(10);
+        }
+
+        private TestContext testContextInstance;
+
+        public TestContext TestContext
+        {
+            get
+            {
+                return testContextInstance;
+            }
+            set
+            {
+                testContextInstance = value;
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void LoginTest()
+        {
+            Login("EugenBorisik", "qwerty12345");
+            Thread.Sleep(35484); // Думаю, это все таки EXPLICIT ожидание
+            string CurrentURL = driver.Url;
+            Assert.AreEqual(ExpectedURL, CurrentURL);
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(5));
+            wait.Until(ExpectedConditions.ElementIsVisible(By.ClassName("sign-out")));
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void LogoutTest()
+        {
+            Login("EugenBorisik", "qwerty12345");
+            driver.FindElement(By.ClassName("sign-out")).Click();
+            string CurrentURL = driver.Url;
+            Assert.AreEqual(URL, CurrentURL);
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void LoginTestWithWaiter()
+        {
+            Login("EugenBorisik", "qwerty12345");
+            string CurrentURL = driver.Url;
+            Assert.AreEqual(ExpectedURL, CurrentURL);
+            driver.FindElement(By.Id("officeMenu")).Click();
+            string CurrentOfficeURL = driver.Url;
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(15));
+            wait.PollingInterval = TimeSpan.FromMilliseconds(2700);
+            wait.Until(ExpectedConditions.ElementIsVisible(By.Id("input-search")));
+            IWebElement Search = driver.FindElement(By.Id("input-search"));
+            Assert.AreEqual(OfficeTabURL, CurrentOfficeURL);
+            Assert.IsTrue(Search.Displayed, "Search input is not displayed!");
```

Comment:
Try to perform 1 assert per test. In your case it is not critical, because your perform assertions at the end